### PR TITLE
indexer: cp sequence number for tx indices query perf

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

title
also merged the schema files of tx indices for simplicity

## Test Plan 

local cargo run to make sure that the column can be populated properly

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by bmwill — APPROVED** [review]

File: crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    PRIMARY KEY(recipient, tx_sequence_number, cp_sequence_number)
+);
+CREATE INDEX tx_recipients_tx_sequence_number_index ON tx_recipients (tx_sequence_number ASC, cp_sequence_number ASC);
+
+CREATE TABLE tx_input_objects (
+    cp_sequence_number          BIGINT       NOT NULL,
+    tx_sequence_number          BIGINT       NOT NULL,
+    -- Object ID in bytes. 
+    object_id                   BYTEA        NOT NULL,
+    PRIMARY KEY(object_id, tx_sequence_number, cp_sequence_number)
```
Comment (wlmyng) [inline crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql:24]: q: how come some of these `tx_` tables have an index, and others don't?

File: crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
+CREATE TABLE tx_senders (
+    cp_sequence_number          BIGINT       NOT NULL,
+    tx_sequence_number          BIGINT       NOT NULL,
+    -- SuiAddress in bytes.
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(sender, tx_sequence_number, cp_sequence_number)
+);
+CREATE INDEX tx_senders_tx_sequence_number_index ON tx_senders (tx_sequence_number ASC, cp_sequence_number ASC);
```
Comment (wlmyng) [inline crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql:8]: when issuing a query, will we need to include `order by cp_sequence_number asc` or `desc` in order to leverage this index on scan or scan backward?

File: crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql (hunk 1 above)
Comment (gegaowp) [reply crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql:24]: those indices were added for tx_reader, others count on the primary key.

File: crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql (hunk 2 above)
Comment (gegaowp) [reply crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql:8]: this index can be used for both forward and backward scans.
https://www.postgresql.org/docs/current/indexes-ordering.html
> The index can also be scanned backward ...

in fact, it's a bit verbose here to specify `ASC` as it's the default one, I will remove it in a following PR I am working on

**Pull request merged**